snap/info.go

```diff
@@ -1453,9 +1453,19 @@ func (app *AppInfo) SecurityTag() string {
 func (app *AppInfo) DesktopFile() string {
 	desktopFileIDs, err := app.Snap.DesktopPlugFileIDs()
 	if err != nil || len(desktopFileIDs) == 0 {
-		// fallback to a simple heuristic "$PREFIX_$APP.desktop"
-		return filepath.Join(dirs.SnapDesktopFilesDir, fmt.Sprintf("%s_%s.desktop", app.Snap.DesktopPrefix(), app.Name))
+		return app.fallbackDesktopFile()
 	}
+
+	if app.CommonID != "" {
+		desktopID := strings.TrimSuffix(app.CommonID, ".desktop")
+
+		if !strutil.ListContains(desktopFileIDs, desktopID) {
+			return app.fallbackDesktopFile()
+		}
```

Comment:
What about the case where an app defines a `common-id` but no `desktop-file-ids`, this feels like a bug, it would make sense to use the desktop file matching the `common-id` still, no?

I suggest the following precedence order:
- try to find file matching `common-id`
- try to find file matching one-of `desktop-file-ids`
- fallback to `$PREFIX_$APP.desktop`

@pedronis @jhenstridge what do you think?